File: CloudDataProtection/CloudDataProtection.Functions.BackupDemo/Triggers/Dto/Result/FileInfoResult.cs
using System.Collections.Generic;
using CloudDataProtection.Functions.BackupDemo.Entities;

namespace CloudDataProtection.Functions.BackupDemo.Triggers.Dto
{
    public class FileInfoResult
    {
        public string Name { get; set; }

        public long Bytes { get; set; }
        
        public string ContentType { get; set; }
        
        public List<FileInfoDestinationResultEntry> UploadedTo { get; set; }
    }

    public class FileInfoDestinationResultEntry
    {
        public int FileDestination { get; set; }
        public string Description { get; set; }

        public FileInfoDestinationResultEntry(FileDestinationInfo info)
        {
            FileDestination = (int) info.Destination;
            Description = info.ToString();
        }
    }
}

File: cloud-data-protection/src/services/result/demo/fileInfoResult.ts
import FileDestination from "entities/fileDestination";

export interface FileInfoResult {
    bytes: number;
    url: string;
    name: string;
    contentType: string;
    uploadedTo: FileInfoDestinationResultEntry[];
}

export interface FileInfoDestinationResultEntry {
    fileDestination: FileDestination;
    description: string;
}